crucible/from-file-json-manipulation-api/run.sh

```diff
@@ -0,0 +1,200 @@
+#!/bin/bash
+
+# user configurable options #####################################################
+
+target="centosstream9-rt"
+
+userenv="default"
+
+#osruntime="chroot"
+osruntime="podman"
+stress_ng_osruntime="chroot"
+
+tools="no"
+#tools="yes"
+
+#stress_ng="no"
+stress_ng="yes"
+
+samples=3
+
+mode="run"
+#mode="test"
+
+from_file="oslat.json"
+
+#################################################################################
+
+DO_JQ_FILENAME="${from_file}.run"
+cp "${from_file}" "${DO_JQ_FILENAME}"
+
+# jq interface that handles the redirection to a temporary file and
+# then replacement of the original file
+# all the functions below use this to avoid having to replicate that
+# file handling logic over and over again
+function do_jq() {
+    jq "$@" "${DO_JQ_FILENAME}" > "${DO_JQ_FILENAME}.tmp"
+    if [ $? == 0 ]; then
+	mv "${DO_JQ_FILENAME}.tmp" "${DO_JQ_FILENAME}"
+	return 0
+    else
+	return 1
+    fi
+}
+
+# change an existing run-params parameter value
+function override_run_param() {
+    local param=$1; shift
+    local value=$1; shift
+
+    do_jq --arg param "${param}" --arg value "${value}" '."run-params"[$param] = $value'
+}
```

Comment:
Different data types can require different syntax, so we may have to do things like this:

```
function override_run_param() {
    local type=$1; shift
    local param=$1; shift
    local value=$1; shift

    case "${type}" in
        "string")
            do_jq --arg param "${param}" --arg value "${value}" '."run-params"[$param] = $value'
            ;;
        "int")
            do_jq --arg param "${param}" --argjson value "${value}" '."run-params"[$param] = $value'
            ;;
    esac
}
```